### ISSUE TITLE
How to generate only selective screenshots with Snapshot

### ISSUE THREAD

**Issue by grzegorzkrukowski** [fastlane/fastlane#16065]
### Question Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Question Subject
I have a question about Snapshot tool.
We have a working setup and we generate screenshots on external servers.
It all works, but with amount of screenshots we need to generate for QA purposes, we are looking for an option to generate just selective tests.

We have:
```
@interface SnapshotUITests : XCTestCase
- (void)testPurchaseFlow;
- (void)testOnboarding;
- (void)testTrainingFlow;
```

From what we found in documentation, snapshot only allows to run all tests from given XCTestCase.

`xcodebuild` itself is allowing to use `-only-testing:TestBundle/TestSuite/TestCase` as parameter to run single test only from all set of tests.

Is it possible to pass and use this parameter via Snapshot ?

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.15.3                                                        |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.21.1 (Apple Git-122.3)                           |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.15.3 (19D76)                                       |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 11.3.1                                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.142.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                    | Version | Update-Status |
| ------------------------- | ------- | ------------- |
| fastlane-plugin-appcenter | 1.8.0   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.3.0        |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.3        |
| claide                    | 1.0.3        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.14.0       |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.6.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.5.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.7.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.7.0        |
| babosa                    | 1.0.3        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.71.0       |
| unf_ext                   | 0.0.7.6      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday                   | 0.17.1       |
| faraday_middleware        | 0.13.1       |
| fastimage                 | 2.1.7        |
| gh_inspector              | 1.1.3        |
| mini_magick               | 4.9.5        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.3.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.7.5        |
| bundler                   | 2.0.2        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.6        |
| jwt                       | 2.1.0        |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| mini_mime                 | 1.0.2        |
| signet                    | 0.12.0       |
| os                        | 1.0.1        |
| googleauth                | 0.9.0        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.36.4       |
| digest-crc                | 0.4.1        |
| google-cloud-storage      | 1.16.0       |
| emoji_regex               | 1.0.1        |
| CFPropertyList            | 3.0.2        |
| tty-cursor                | 0.7.0        |
| tty-spinner               | 0.9.2        |
| json                      | 2.3.0        |
| multi_json                | 1.14.1       |
| memoist                   | 0.16.2       |
| google-cloud-env          | 1.3.0        |
| google-cloud-core         | 1.4.1        |
| forwardable               | 1.2.0        |
| logger                    | 1.3.0        |
| stringio                  | 0.0.2        |
| ipaddr                    | 1.2.2        |
| openssl                   | 2.1.2        |
| ostruct                   | 0.1.0        |
| strscan                   | 1.0.0        |
| date                      | 2.0.0        |
| fileutils                 | 1.1.0        |
| etc                       | 1.0.1        |
| io-console                | 0.4.7        |
| zlib                      | 1.0.0        |
| libxml-ruby               | 3.1.0        |
| rexml                     | 3.1.9        |
| psych                     | 3.1.0        |
| mutex_m                   | 0.1.0        |
| fastlane-plugin-appcenter | 1.8.0        |
| csv                       | 3.0.9        |
</details>

*generated on:* **2020-02-21**
</details>

**Comment by janpio** [comment]
That sounds like a reasonable wish.

`scan, the sister action of `snapshot` seems to support an `only_testing` option: https://docs.fastlane.tools/actions/scan/#parameters https://github.com/fastlane/fastlane/search?q=only_testing&type=Code You can probably quite easily adapt that for `snapshot` as well and submit a Pull Request.

**Comment by grzegorzkrukowski** [comment]
Thanks for answer.

For now I found out that it's possible to pass `xcargs` directly to `capture_ios_screenshots` lane.
I am using it to pass manually generated `-only-testing:UITests/UITests/test` string to `xcodebuild`.
I was able to achieve what I needed this way. I found it when looking at tests of `snapshot`.

Probably it will be a good idea to have separate support just for `-only-testing` anyways and document it. I will add support for it soon.
